### DISCUSSION TITLE
invalidate and loading state for FutureProvider

### DISCUSSION THREAD

**Discussion by benny856694** [rrousselGit/riverpod#2821] [General]
I want to invalidate a FutureProvider, and I'd like the loading callback to be called on invalidating, 
How can I achieve this? 
so far, it won't be called when invalidated.

**Comment by AhmedLSayed9** [comment]
You can set `skipLoadingOnRefresh` to true at .when, or use Dart 3's [switch-case](https://github.com/rrousselGit/riverpod/issues/2715)

**Reply by benny856694** [reply]
works like a charm, thanks.
